cli-plugins/apps/getting-started/configuration.hbs.md

```diff
@@ -0,0 +1,168 @@
+consumption of resources when uploading the source.
+
+When iterating on code with the `--live-update` flag enabled, changes to directories or files
+listed in `.tanzuignore` do not trigger the automatic re-deployment of the source code.
+
+The following are some guidelines for the `.tanzuignore` file:
+
+- The `.tanzuignore` file should include a reference to itself, as it provides no value when deployed.
+- Directories must not end with the system separator `/`, or `\`.
+- Comments using hashtag `#` can be included.
+- If the `.tanzuignore` file contains files or directories that are not found in the source code, they are ignored.
+
+### Example of a .tanzuignore file
+
+```console
+.tanzuignore # must contain itself in order to be ignored
+# This is a comment
+this/is/a/folder/to/exclude
+
+this-is-a-file.ext
+```
+
+## Registry flags and environment variables
+
+The Apps CLI plug-in allows users to push images to their private registry by setting some flags 
+when creating workloads from local source code.
+
+A user can either trust a custom certificate on their system or pass the path to the certificate via flags.
+
+To pass the certificate through flags, specify:
+
+- `--registry-ca-cert`: This is the path of the self-signed certificate needed for the custom or 
+  private registry. This is also populated with a default value through the environment variable `TANZU_APPS_REGISTRY_CA_CERT`.
+- `--registry-password`: Use this when the registry requires credentials to push. The value of this 
+  flag can also be specified through `TANZU_APPS_REGISTRY_PASSWORD`.
+- `--registry-username`: Use with `--registry-password` to set the registry credentials. It can also
+  be provided as the environment variable `TANZU_APPS_REGISTRY_USERNAME`.
+- `--registry-token`: Set when the registry authentication is done through a token. The value of this
+  flag can also be taken from the `TANZU_APPS_REGISTRY_TOKEN` environment variable.
+
+For example:
+
+```bash
+tanzu apps workload apply WORKLOAD \
+--local-path PATH-TO-REPO -s registry.url.nip.io/PACKAGE/IMAGE \
+--type web --registry-ca-cert path/to/ca/cert.nip.io.crt \
+--registry-username USERNAME \
+--registry-password PASSWORD
+```
+
+Alternatively, the same command can be run as:
+
+```bash
+export TANZU_APPS_REGISTRY_CA_CERT=path/to/ca/cert.nip.io.crt
+export TANZU_APPS_REGISTRY_PASSWORD=USERNAME
+export TANZU_APPS_REGISTRY_USERNAME=PASSWORD
+
+tanzu apps workload apply WORKLOAD \
+--local-path PATH-TO-REPO \
+-s registry.url.nip.io/PACKAGE/IMAGE
+```
+
+`--type` is a flag that can be set through environment variables as well so it’s applied to all the
+workloads to be created. If the workload type is not set as an environment variable or it’s not set
+as a flag either, then it will be defaulted to `web`.
+
+```bash
+export TANZU_APPS_TYPE=server
+```
+
+## <a id='autocompletion'></a>Autocompletion
+
+The Apps CLI plug-in has auto-completion support. The plug-in supports auto-completion for commands,
```

Comment:
The Apps CLI plug-in provides auto-completion support for commands,
positional arguments, flags, and flag values. 

Add one of the following commands to the shell config file according to your setup:

Reply:
updated